proto/v1beta1/spdx.proto

```diff
@@ -0,0 +1,391 @@
+  string summary_description = 13;
+  // A more detailed description of the package
+  string detailed_description = 14;
+
+  // An External Reference allows a Package to reference an external source of
+  // additional information, metadata, enumerations, asset identifiers, or
+  // downloadable content believed to be relevant to the Package
+  message ExternalRef {
+    // The category of the external reference
+    enum Category {
+      // Unspecified
+      CATEGORY_UNSPECIFIED = 0;
+      // Security (e.g. cpe22Type, cpe23Type)
+      SECURITY = 1;
+      // Package Manager (e.g. maven-central, npm, nuget, bower, purl)
+      PACKAGE_MANAGER = 2;
+      // Persistent-Id (e.g. swh)
+      PERSISTENT_ID = 3;
+      // Other
+      OTHER = 4;
+    }
+
+    // An External Reference allows a Package to reference an external source of
+    // additional information, metadata, enumerations, asset identifiers, or
+    // downloadable content believed to be relevant to the Package
+    Category category = 1;
+    // Type of category (e.g. 'npm' for the PACKAGE_MANAGER category)
+    string type = 2;
+    // The unique string with no spaces necessary to access the package-specific
+    // information, metadata, or content within the target location
+    string locator = 3;
+    // Human-readable information about the purpose and target of the reference
+    string comment = 4;
+  }
+  // ExternalRef
+  repeated ExternalRef external_refs = 15;
+  // A place for the SPDX data creator to record, at the package level,
+  // acknowledgements that may be needed to be communicated in some contexts
+  string attribution = 16;
+}
+
+// PackageInfoOccurrence represents an SPDX Package Information
+// section: https://spdx.github.io/spdx-spec/3-package-information/
+message PackageInfoOccurrence {
+  // Uniquely identify any element in an SPDX document which may be referenced
+  // by other elements
+  string id = 1;
+  // Provide the actual file name of the package, or path of the directory being
+  // treated as a package
+  string filename = 2;
+  // Provide a place for the SPDX file creator to record any relevant background
+  // information or additional comments about the origin of the package
+  string source_info = 3;
+  // package or alternative values, if the governing license cannot be
+  // determined
+  // TODO: create a license proto to encapsulate the repeating license fields
+  string license_concluded = 4;
+  // This field provides a place for the SPDX file creator to record any
+  // relevant background information or analysis that went in to arriving at the
+  // Concluded License for a package
+  string license_comments = 5;
+  // A place for the SPDX file creator to record any general
+  // comments about the package being described
+  string comment = 6;
+}
+
+// RelationshipNote represents an SPDX Relationship
+// section:
+// https://spdx.github.io/spdx-spec/7-relationships-between-SPDX-elements/
+message RelationshipNote {}
```

Comment:
Having an empty Relationship Note is a bit strange given the model of Notes and Occurrences we've been using. Do you have an example in mind for a pair of SPDX elements and relationship between them and how to represent them?

Reply:
I added the Relationship type to the note to not leave it completely empty. An example of a SPDX relationship is a package A depends on a package B. In this case the source and target of a relationship would be a PackageInfoOccurrence.